src/main/java/com/fakng/fakngagrgtr/FakngScheduler.java

```diff
@@ -0,0 +1,69 @@
+package com.fakng.fakngagrgtr;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Stream;
+
+import javax.annotation.PostConstruct;
+
+import com.fakng.fakngagrgtr.parser.Parser;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import com.fakng.fakngagrgtr.persistent.vacancy.VacancyRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class FakngScheduler {
+
+    private final List<Parser> parsers;
+    private final VacancyRepository vacancyRepository;
+    @Value("${agrgtr.pool-size:1}")
+    private int poolSize;
+    private ExecutorService executor;
+
+    @PostConstruct
+    public void init() {
+        executor = Executors.newFixedThreadPool(poolSize);
+    }
+
+    @Scheduled(cron = "${agrgtr.cron}:0 */12 * * * *")
+    public void scheduleAggregation() {
+        parsers.forEach(parser -> CompletableFuture
+                .supplyAsync(parser::parse)
+                .thenAccept(this::updateCompanyVacancies)
+                .exceptionally(ex -> {
+                    ex.printStackTrace();
+                    return null;
+                }));
+    }
+
+    private void updateCompanyVacancies(List<Vacancy> actualVacancies) {
+        if (actualVacancies.isEmpty()) {
+            return;
+        }
+        List<Vacancy> dbVacancies = vacancyRepository.findByCompany(
+                actualVacancies.get(0).getCompany());
+        removeOutdatedVacancies(actualVacancies, dbVacancies);
+        addNewVacancies(actualVacancies, dbVacancies);
+    }
+
+    private void removeOutdatedVacancies(List<Vacancy> actualVacancies, List<Vacancy> dbVacancies) {
+        List<Vacancy> outdatedVacancies = dbVacancies.stream()
+                .filter(vacancy -> !actualVacancies.contains(vacancy))
```

Comment:
This is bad. Contains in list is O(n) operation
That means that if I have a list of N actual vacancies and list of M dbVacancies, the complexity of this method becomes O(N * M)

So if there are ~5k vacancies from both sides, we'll be doing ~25k checks

Even if we decide to use this approach (though I'd estimated the performance of others also), we should use HashSet or HashMap instead of list for O(1) checks of presence

Reply:
I got the approach, but we need to carefully design this one and also look at other possible options. Probably estimate 2-3 designs from mem usage, amount of DB queries on avg, etc. This would be wise. 

I like this one though, we can keep it as a draft for now. But others should also be considered at least on paper. And for this one, we need some avg numbers of metrics I mentioned above.